debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java

```diff
@@ -98,6 +98,28 @@
      * @return the connection factory
      */
     public static ConnectionFactory patternBasedFactory(String urlPattern, Field... variables) {
+        return patternBasedFactory(urlPattern, null, variables);
+    }    
+    
+    /**
+     * Create a {@link ConnectionFactory} that replaces variables in the supplied URL pattern. Variables include:
+     * <ul>
+     * <li><code>${hostname}</code></li>
+     * <li><code>${port}</code></li>
+     * <li><code>${dbname}</code></li>
+     * <li><code>${username}</code></li>
+     * <li><code>${password}</code></li>
+     * </ul>
+     * 
+     * @param urlPattern the URL pattern string; may not be null
+     * @param variables any custom or overridden configuration variables
+     * @param classloader When the driver class name is null, the DriverManager will resolve the driver using the URL pattern. 
+     *        Otherwise, the this method will attempt to load the driver class (optionally using the classloader if one is 
+     *        provided), create a new instance of the Driver with the URL and properties, and then attempt to connect. 
+     * @return the connection factory
+     */
+    @SuppressWarnings("unchecked")
```

Comment:
I still don't think the JavaDoc really does a good job explaining what this method does, since you have to read the `classloader` parameter description to really know what's going to happen. And the order of the parameters still is not right. Please replace the whole JavaDoc for this method with the following:

```
    /**
     * Create a {@link ConnectionFactory} that uses the specific JDBC driver class loaded with the given class loader, and obtains the connection URL by replacing the following variables in the URL pattern:
     * <ul>
     * <li><code>${hostname}</code></li>
     * <li><code>${port}</code></li>
     * <li><code>${dbname}</code></li>
     * <li><code>${username}</code></li>
     * <li><code>${password}</code></li>
     * </ul>
     * <p>
     * This method attempts to instantiate the JDBC driver class and use that instance to connect to the database.
     * @param urlPattern the URL pattern string; may not be null
     * @param driverClassName the name of the JDBC driver class; may not be null
     * @param classloader the ClassLoader that should be used to load the JDBC driver class given by `driverClassName`; may be null if this class' class loader should be used
     * @param variables any custom or overridden configuration variables
     * @return the connection factory
     */
```